opentelemetry/src/global/internal_logging.rs

```diff
@@ -108,7 +114,28 @@ macro_rules! otel_error {
     (name: $name:expr, $($key:ident = $value:expr),+ $(,)?) => {
         #[cfg(feature = "internal-logs")]
         {
-            tracing::error!(name: $name, target: env!("CARGO_PKG_NAME"), $($key = $value),+, "");
+            tracing::error!(name: $name,
+                            target: env!("CARGO_PKG_NAME"),
+                            $($key = {
+                                    $crate::format_value!($value)
+                            }),+,
+                            ""
+                    )
         }
     };
 }
+
+/// Helper macro to format a value using Debug if available, falling back to Display
+#[macro_export]
+macro_rules! format_value {
```

Comment:
This avoid calling format() explicitly by the caller.

Reply:
unsure of this.. does format! make every attribute value a stringified version?

Reply:
Please do these changes separately. I am not sure if this is needed, as this is making the "thin" wrapper not so thin.

Reply:
Without this, we need to stringy each attribute at the caller site.

Reply:
does users of `tracing` have to it themselves? If yes, we (i.e all otel components doing this logging) can do the same.

Also, I don't think that every attribute should be stringified. There can be integer attributes like item_count, floating attributes like sampling_ratio etc. By stringifying in the helper, we lose the type information for everyone, even when they did not intend to do so.

Reply:
> does users of tracing have to it themselves? If yes, we (i.e all otel components doing this logging) can do the same.

yes, tracing crate has to do it themself. 

> By stringifying in the helper, we lose the type information for everyone, even when they did not intend to do so.

I didn't get this ?

Reply:
By stringifying all attributes, won't it always result getting serialized as string, even when user provide an integer/float/bool value?
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-appender-tracing/src/layer.rs#L86

Reply:
Make sense, have removed the formatting .